TowerForge/TowerForge/GameModule/Events/EventTransformation.swift

```diff
@@ -0,0 +1,10 @@
+//
+//  EventTransformation.swift
+//  TowerForge
+//
+//  Created by Keith Gan on 31/3/24.
+//
+
+protocol EventTransformation {
+    func transformEvent(event: TFEvent) -> TFEvent
```

Comment:
I think it's better for transformEvent to return an optional so that transforms can return nil events as needed. One example could be the invulnerable event. If the transform returns nil, we can stop all further transforms on it and skip and move on to the next event.

Reply:
I added a `DisabledEvent` which I think will avoid the optional value.